docs/guides/non-custodial wallets/issue-a-virtual-card.md

```diff
@@ -24,17 +24,72 @@ A user will more often than not be quoted a price for a purchase by a merchant i
 
 In order to ensure that funds are sufficiently locked within the [smart contract](/contracts/payment-protocol) such that Immersve are in a position to approve an authorization request received via the card scheme network you will need to have firstly locked sufficient digital assets within the smart contract via digital asset transfer and smart contract invocations as instructed by the [get prerequisite transactions](/api-reference/get-prerequisites) operation. The necessary blockchain transactions are contained within the `requiredTransactions` collection returned.
 
-If the user has not transacted using the solution before then the the [get prerequisite transactions](/api-reference/get-prerequisites) response will typically call for and `erc20_approval` in favour of the smart contract followed by a `depositAndCreateLockedFund` invocation of the smart contract.
+If the user has not transacted using the solution before then the the [get prerequisite transactions](/api-reference/get-prerequisites) response will typically call for and ERC20 `approve` in favour of the smart contract followed by a `depositAndLock` invocation of the smart contract.
 
 If multiple transactions are present then they should be carried out in the order in which they are presented.
 
 The client application is to parse and formulate the raw blockchain transaction message such that it is presented to the user within the web3 wallet and signed and sent to the relevant blockchain.
 
+#### How to use prerequisites example for ts/js
+
+[Prerequisites](/api-reference/get-prerequisites) endpoint returns Array of actions to be performed.  
+Each object in array has `type` and `params`.  
+Type `smart_contract_write` means that this action is about interaction with the smart contract.  
+`params` contain all details required to perform this interaction.
+
+<details>
+<summary>Code snippet (using ethers.js)</summary>
+
+Typescript code
+
+```ts
+import { JsonFragment } from '@ethersproject/abi';
+/**
+ * Builds array of params to be used in calls to smart contracts
+ * @param abi ABI for smart contact function
+ * @param method function name to be called
+ * @param params object of properties matching to the input names specified in ABI
+ * @returns array of params in correct order
+ */
+export function toSmartContractParams(
```

Comment:
I think this function is not required; just return `Object.values(params)`

Reply:
This function gives confidence in order of the params

Reply:
found this https://stackoverflow.com/questions/5525795/does-javascript-guarantee-object-property-order and decided to make sure order is correct

Reply:
We need to make our APIs easy to use. The order should be guaranteed on the server not on the client.

Reply:
removed this block in favour of 
```ts
Object.values(params)
```